pkg/oc/cli/admin/migrate/migrator.go

```diff
@@ -143,36 +178,36 @@ func (o *ResourceOptions) Bind(c *cobra.Command) {
 	c.Flags().StringVar(&o.FromKey, "from-key", o.FromKey, "If specified, only migrate items with a key (namespace/name or name) greater than or equal to this value")
 	c.Flags().StringVar(&o.ToKey, "to-key", o.ToKey, "If specified, only migrate items with a key (namespace/name or name) less than this value")
 
-	// kcmdutil.PrinterForCommand needs these flags, however they are useless
-	// here because oc process returns list of heterogeneous objects that is
-	// not suitable for formatting as a table.
-	kcmdutil.AddNonDeprecatedPrinterFlags(c)
+	o.PrintFlags.AddFlags(c)
 
 	usage := "Filename, directory, or URL to docker-compose.yml file to use"
 	kcmdutil.AddJsonFilenameFlag(c.Flags(), &o.Filenames, usage)
 }
 
 func (o *ResourceOptions) Complete(f kcmdutil.Factory, c *cobra.Command) error {
 	o.Output = kcmdutil.GetFlagString(c, "output")
+
+	if o.DryRun {
+		o.PrintFlags.Complete("%s (dry run)")
```

Comment:
?

Reply:
This looks weird to me as well, but it is the current solution that we have in place for generating dry-run messages.

By default, if no `--output` argument is supplied by a user, the printing stack's `ToPrinter()` method returns a `success-message` printer, which as the name implies, prints a successful message (such as `pod/foo-bar successfully deleted` or something similar). In the case of a dry-run for the command, the only way to currently note that in a successful output (using the printing stack) is to go through the flags' Complete method.